src/main/java/io/supertokens/webserver/api/dashboard/DashboardUserAPI.java

```
@@ -0,0 +1,207 @@
+            // normalize new email
+            newEmail = normalizeStringParam(newEmail, "newEmail");
+
+            // check if the newEmail is in valid format
+            if (!Dashboard.isValidEmail(newEmail)) {
+                JsonObject response = new JsonObject();
+                response.addProperty("status", "INVALID_EMAIL_ERROR");
+                super.sendJsonResponse(200, response, resp);
+                return;
+            }
+        }
+
+        String newPassword = InputParser.parseStringOrThrowError(input, "newPassword", true);
+        if (newPassword != null) {
+            // normalize new password
+            newPassword = normalizeStringParam(newPassword, "newPassword");
+            // check if the new password is strong
+            if (Dashboard.isStrongPassword(newPassword)) {
+                JsonObject response = new JsonObject();
+                response.addProperty("status", "PASSWORD_WEAK_ERROR");
+                super.sendJsonResponse(200, response, resp);
+                return;
+            }
+        }
+
+        try {
+            String userId = InputParser.parseStringOrThrowError(input, "userId", true);
+            if (userId != null) {
+                // normalize userId
+                userId = normalizeStringParam(userId, "userId");
+
+                Dashboard.updateUsersCredentialsWithUserId(main, userId, newEmail, newPassword);
+                JsonObject response = new JsonObject();
+
+                response.addProperty("status", "OK");
+                super.sendJsonResponse(200, response, resp);
+            }
+
+            String email = InputParser.parseStringOrThrowError(input, "email", true);
+            if (email != null) {
+                // normalize userId
+                email = normalizeStringParam(email, "email");
+                Dashboard.updateUsersCredentialsWithEmail(main, email, newEmail, newPassword);
+
+                JsonObject response = new JsonObject();
```

The API should throw a 400 if both userId and email are missing